Scalar/CommandLine/CloneVerb.cs

```diff
@@ -393,8 +389,28 @@ private Result GitClone()
                 return new Result($"Failed to complete regular clone: {fetchResult?.Errors}");
             }
 
-            GitProcess.Result checkoutResult = null;
+            // Configure the specified branch, or the default branch on the remote if not specified
+            string branch = this.Branch;
+            if (branch is null)
+            {
+                if (!git.TryGetRemoteDefaultBranch("origin", out branch, out string defaultBranchError))
+                {
```

Comment:
nit: these could be grouped with `&&` since there is no `else` for the inner `if`.